File: from-2701-to-2800/2779-maximum-beauty-of-an-array-after-applying-operation/solution.go
func maximumBeauty(nums []int, k int) int {
    sort.Ints(nums)
    result := 0
    left := 0
    for right := 0; right < len(nums); right ++ {
        for nums[right] - nums[left] > 2 * k {
            left += 1
        }
        curLen := right - left + 1
        if curLen > result {
            result = curLen
        }
    }
    return result
}

File: from-2701-to-2800/2779-maximum-beauty-of-an-array-after-applying-operation/solution.py
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort()
        result = 0
        left = 0
        for right in range(len(nums)):
            while nums[right] - nums[left] > 2 * k:
                left += 1

            result = max(result, right - left + 1)

        return result